content/docs/fundamentals/converting/resolving-errors@zh_CN.md

```diff
@@ -0,0 +1,350 @@
+
+将以下代码**添加**为 `<head>` 标记的第一行代码：
+
+
+```html
+<meta charset="utf-8" />
+```
+
+**保存**文件并重新加载网页。验证字符集错误是否不再出现。
+
+## 包含权威链接
+
+现在，我们来看看以下错误：
+
+<pre class="error-text">
+必需的标记“link rel=canonical”缺失或不正确。
+</pre>
+
+每个 AMP 文档都必须拥有引用此文档“权威”版本的链接。在本教程的[使您的网页可被轻松发现](/zh_cn/docs/fundamentals/converting/discoverable.html)步骤中，我们将详细了解权威网页的定义，以及实现权威关联的不同方法。
+
+对于本教程，我们将要转换的原始 HTML 文章视为权威网页。
+
+请将以下代码**添加**到 `<meta charset="utf-8" />` 标记下方：
+
+```html
+<link rel="canonical" href="/article.html">
+```
+
+{% call callout('注意', type='note') %}
+您可以创建独立的权威 AMP 网页。在这种情况下，权威链接仍是必需的，但应指向 AMP 文章本身：
+
+```html
+<link rel="canonical" href="article.amp.html">
+```
+{% endcall %}
+
+现在，**重新加载**网页。虽然仍有很多错误需要修正，但权威链接错误已不复存在。
+
+## 指定 AMP 属性
+
+AMP 要求网页的根 `<html>` 元素中具有相关属性，以将网页声明为 AMP 文档。
+
+<pre class="error-text">
+标记“html ⚡ for top-level html”中缺少必需的属性“⚡”
+必需的标记“html ⚡ for top-level html”缺失或不正确。
+</pre>
+
+只需将 `⚡` 属性添加到 `<html>` 标记，即可解决上述错误，如下所示：
+
+```html
+<html ⚡ lang="en">
+```
+
+现在，请重新加载网页并检查这两个错误是否已消失。
+
+{% call callout('注意', type='note') %}
+虽然建议您指定 `⚡`，但您也可以使用 `amp` 属性来替代 `⚡` 属性，如下所示：
+
+```html
+<html amp lang="en">
+```
+{% endcall %}
+
+## 指定视口
+
+接下来，我们将解决以下错误：
+
+<pre class="error-text">
+必需的标记“meta name=viewport”缺失或不正确。
+</pre>
+
+AMP 要求您为视口定义 `width` 和 `minimum-scale`。这两个值必须分别定义为 `device-width` 和 `1`。视口是 HTML 网页的 `<head>` 中包含的常用标记。
+
+要解决视口错误，请将以下 HTML 代码段添加到 `<head>` 标记中：
+
+```html
+<meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
+```
+
+在 AMP 中，必需为 `width` 和 `minimum-scale` 指定值。定义 `initial-scale` 不是一项强制性要求，但移动网站开发者通常都会指定此值，所以我们建议您也这么做。要详细了解视口和自适应设计，请参阅[配置视口](https://developers.google.com/speed/docs/insights/ConfigureViewport)。
+
+和之前一样，**重新加载**网页并检查错误是否已消失。
+
+## 替换外部样式表
+
+以下错误与使用样式表有关：
+
+<pre class="error-text">
+标记“link rel=stylesheet for fonts”中的属性“href”设为了无效值“base.css”。
+</pre>
+
+具体来说，该错误涉及的是 `<head>` 标记中的下列样式表链接标记：
+
+```html
+<link href="base.css" rel="stylesheet" />
+```
+
+问题在于，这是一个外部样式表引用。在 AMP 网页中，为尽可能缩短文档的加载用时，您不能添加外部样式表。相反，您必须使用 `<style amp-custom></style>` 标记在 AMP 文档中以内嵌的方式添加所有样式表规则。
+
+```html
+<style amp-custom>
+
+/* The content from base.css */
+
+</style>
+```
+
+接下来，我们来解决这个错误：
+
+1.  **移除** `<head>` 中指向外部样式表的 `<link>` 标记，然后将其替换为内嵌的 `<style amp-custom></style>` 标记。样式标记中的 `amp-custom` 属性必需提供。
+2. 将 [`base.css`](https://github.com/googlecodelabs/accelerated-mobile-pages-foundations/blob/master/base.css) 文件中的所有样式**复制**到 `<style amp-custom></style>` 标记中。
+
+和之前一样，**重新加载**网页并验证样式表错误是否已消失。
+
+{% call callout('注意', type='note') %}
+除了必须采用内嵌样式以外，所有样式信息的文件大小还不能超过 50 千字节的上限。您应该先使用 [SASS](http://sass-lang.com/) 等 CSS 预处理器压缩您的 CSS，然后再将 CSS 内嵌到 AMP 网页中。
+{% endcall %}
+
+{% call callout('重要提示', type='caution') %}
+整个 AMP 文档中只能有 1 个样式标记。如果 AMP 网页引用了多个外部样式表，则您需要将这些样式表整理为一组规则。要了解哪些 CSS 规则在 AMP 中有效，请参阅[受支持的 CSS](/zh_cn/docs/design/responsive/style_pages.html)。
+{% endcall %}
+
+## 排除第三方 JavaScript
+
+虽然对于 AMP 网页，您可以通过内嵌 CSS 的方式相对轻松地重新制作样式表，但 JavaScript 却并非如此。
+
+<pre class="error-text">
```

Comment:
All of these code blocks need to be in EN, so please update with EN translation (https://www.ampproject.org/docs/fundamentals/converting/resolving-errors).